requests/adapters.py

```diff
@@ -300,7 +300,7 @@ def get_connection(self, url, proxies=None):
         proxy = select_proxy(url, proxies)
 
         if proxy:
-            proxy = prepend_scheme_if_needed(proxy, 'http')
+            proxy = prepend_scheme_if_needed(proxy, 'http').strip()
```

Comment:
We may want to call `strip` on the proxy itself before handing it to `prepend_scheme_if_needed`. Leading whitespace causes issues with `urlparse` and we get back a weird url which will fail later.

Reply:
I'm not even certain either of those are the right change, honestly, However, I can't think of a better change.

Reply:
I agree this isn’t great, but I’m not sure if we have a better place to catch both environmental and parameter proxies for sanitizing. We’d need to make a larger change to how proxies are handled I think if we wanted this to be done in a less hacky fashion.

Reply:
Do you mean to do something like this?
```python
if proxy:
    proxy.strip()
```